File: graffiti/src/main/java/com/ue/graffiti/util/ResourceUtils.java
package com.ue.graffiti.util;

import android.content.Context;
import android.content.res.TypedArray;

/**
 * Created by hawk on 2018/1/17.
 */

public class ResourceUtils {

    public static int[] getImageArray(Context context, int arrayId) {
        TypedArray ta = context.getResources().obtainTypedArray(arrayId);
        int[] imageArray = new int[ta.length()];
        for (int i = 0, len = ta.length(); i < len; i++) {
            imageArray[i] = ta.getResourceId(i, 0);
        }
        ta.recycle();
        return imageArray;
    }
}

File: graffiti/src/main/java/com/ue/graffiti/util/ResourceUtils.kt
package com.ue.graffiti.util

import android.content.Context

/**
 * Created by hawk on 2018/1/17.
 */

object ResourceUtils {
    fun getImageArray(context: Context, arrayId: Int): IntArray {
        val ta = context.resources.obtainTypedArray(arrayId)
        val imageArray = IntArray(ta.length())
        for (i in imageArray.indices) {
            imageArray[i] = ta.getResourceId(i, 0)
        }
        ta.recycle()
        return imageArray
    }
}
